### PR TITLE
pubkey_to_privkey_verify

### PR DESCRIPTION
**Pull request by daedalus**
It just verifies if a given pubkey and a given privkey matches

### CODE REVIEW

**Comment by vbuterin** [comment]
I think you might also want to all decode_pubkey to decode the pubkey to the standard tuple format after dividing.

File: bitcoin/main.py [hunk 1]
```diff
@@ -270,6 +276,16 @@ def divide(pubkey, privkey):
     factor = inv(decode_privkey(privkey), N)
     return multiply(pubkey, factor)
 
+def pubkey_to_privkey_verify(pubkey,privkey):
+        f1 = get_pubkey_format(pubkey)
+        p1 = decode_pubkey(pubkey,f1)
+        f2 = get_privkey_format(privkey)
+        p2 = decode_privkey(privkey,f1)
```
Comment (wizardofozzie) [inline bitcoin/main.py]: should it be `p2 = decode_privkey(privkey, f2)`?

**Comment by wizardofozzie** [comment]
I had a similar idea, [see my fork](https://github.com/wizardofozzie/pybitcointools/blob/master/bitcoin/main.py#L55-L58)

though I didn't see the need for a verify function outside unittesting

**Comment by daedalus** [comment]
I believe that having a separate function for point in curve checking doesn't do a harm and provides some better readability to the code.

**Comment by daedalus** [comment]
I also have the idea of separating all functions related to eliptic curves math theory in a diferent module than main. But that is for future.

File: bitcoin/main.py (hunk 1 above)
Comment (daedalus) [inline bitcoin/main.py]: yes, my bad

**Pull request closed without merging**